addons/event/models/event_registration.py

```diff
@@ -181,10 +184,39 @@ def name_get(self):
 
     def _check_auto_confirmation(self):
         if any(not registration.event_id.auto_confirm or
-               (not registration.event_id.seats_available and registration.event_id.seats_limited) for registration in self):
+               (not registration.event_id.seats_available and registration.event_id.seats_limited)
+               for registration in self):
             return False
         return True
 
+    def toggle_active(self):
```

Comment:
Isn't it possible to include it as a trigger on event and ticket constraints (registration_ids.active) ?

Reply:
It was brought to my attention (by odoo-bin) that `@api.constrains` do not support field attributes.

Reply:
Indeed.